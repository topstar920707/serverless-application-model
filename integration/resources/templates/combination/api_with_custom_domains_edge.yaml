Parameters:
  MyEdgeDomainName:
    Type: String
  MyEdgeDomainCert:
    Type: String

Resources:
  MyFunction:
    Type: AWS::Serverless::Function
    Properties:
      InlineCode: |
        exports.handler = async (event) => {
          const response = {
            statusCode: 200,
            body: JSON.stringify('Hello from Lambda!'),
          };
          return response;
        };
      Handler: index.handler
      Runtime: nodejs14.x
      Events:
        Fetch:
          Type: Api
          Properties:
            RestApiId:
              Ref: MyApi
            Method: Get
            Path: /get

  MyApi:
    Type: AWS::Serverless::Api
    Properties:
      OpenApiVersion: 3.0.1
      StageName: Prod
      Domain:
        DomainName:
          Ref: MyEdgeDomainName
        CertificateArn:
          Ref: MyEdgeDomainCert
        EndpointConfiguration: EDGE
        BasePath:
          - /get
        Route53:
          HostedZoneId: Z1SKZDMQ2UR7IW
          IpV6: true