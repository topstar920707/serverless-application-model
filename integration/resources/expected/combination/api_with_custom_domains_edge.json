[
  { "LogicalResourceId":"RecordSetGroup", "ResourceType":"AWS::Route53::RecordSetGroup" },
  { "LogicalResourceId":"MyApiDeployment", "ResourceType":"AWS::ApiGateway::Deployment" },
  { "LogicalResourceId":"ApiGatewayDomainName", "ResourceType":"AWS::ApiGateway::DomainName" },
  { "LogicalResourceId":"MyApi", "ResourceType":"AWS::ApiGateway::RestApi" },
  { "LogicalResourceId":"MyApiProdStage", "ResourceType":"AWS::ApiGateway::Stage" },
  { "LogicalResourceId":"MyFunction", "ResourceType":"AWS::Lambda::Function" },
  { "LogicalResourceId":"MyFunctionRole", "ResourceType":"AWS::IAM::Role" },
  { "LogicalResourceId":"MyApigetBasePathMapping", "ResourceType":"AWS::ApiGateway::BasePathMapping" },
  { "LogicalResourceId":"MyFunctionFetchPermissionProd", "ResourceType":"AWS::Lambda::Permission" }
]